src/services/pcn-router/src/PortsSecondaryip.cpp

```diff
@@ -1,118 +1,54 @@
 
-#include "PortsSecondaryip.h"
-#include "Router.h"
+// TODO: Modify these methods with your own implementation
 
-#include <stdexcept>
 
-PortsSecondaryip::PortsSecondaryip(Ports &parent,
-                                   const PortsSecondaryipJsonObject &conf)
-    : parent_(parent) {
-  ip_ = conf.getIp();
-  netmask_ = conf.getNetmask();
+#include "PortsSecondaryip.h"
+#include "Router.h"
 
-  parent.logger()->debug("Adding secondary address [ip: {0} - netmask: {1}]",
-                         ip_, netmask_);
 
+PortsSecondaryip::PortsSecondaryip(Ports &parent, const PortsSecondaryipJsonObject &conf)
+    : PortsSecondaryipBase(parent) {
   // TODO: check that no other router port exists in the same network
+  ip_ = conf.getIp();
+  parent.logger()->debug("Adding secondary address [ip: {0}]", ip_);
 
   /*
   * Add two routes in the routing table
   */
   int index = parent.parent_.Cube::get_port(parent.getName())->index();
-  parent.parent_.add_local_route(ip_, netmask_, parent.getName(), index);
+  parent.parent_.add_local_route(ip_, parent.getName(), index);
 }
 
 PortsSecondaryip::~PortsSecondaryip() {}
 
-void PortsSecondaryip::update(const PortsSecondaryipJsonObject &conf) {
-  // This method updates all the object/parameter in PortsSecondaryip object
-  // specified in the conf JsonObject.
-  // You can modify this implementation.
-}
-
-PortsSecondaryipJsonObject PortsSecondaryip::toJsonObject() {
-  PortsSecondaryipJsonObject conf;
-
-  conf.setIp(getIp());
-
-  conf.setNetmask(getNetmask());
-
-  return conf;
-}
-
-// TODO: move in the Ports class? Probably it is better
-void PortsSecondaryip::updatePortInDataPath(Ports &parent) {
-  auto router_port =
-      parent.parent_.get_hash_table<uint16_t, r_port>("router_port");
-
-  r_port value{
-      .ip = utils::ip_string_to_be_uint(parent.getIp()),
-      .netmask = utils::ip_string_to_be_uint(parent.getNetmask()),
-      .secondary_ip = {},
-      .secondary_netmask = {},
-      .mac = utils::mac_string_to_be_uint(parent.getMac()),
-  };
-
-  uint16_t index = parent.parent_.get_port(parent.getName())->index();
-  int i = 0;
-  for (auto &addr : parent.getSecondaryipList()) {
-    value.secondary_ip[i] = utils::ip_string_to_be_uint(addr->getIp());
-    value.secondary_netmask[i] =
-        utils::ip_string_to_be_uint(addr->getNetmask());
-    i++;
-  }
-
-  router_port.set(index, value);
+std::string PortsSecondaryip::getIp() {
+  // This method retrieves the ip value.
+  return ip_;
 }
 
 void PortsSecondaryip::createInControlPlane(
-    Ports &parent, const std::string &ip, const std::string &netmask,
+    Ports &parent, const std::string &ip,
     const PortsSecondaryipJsonObject &conf) {
   // This method creates the actual PortsSecondaryip object given thee key
   // param.
 
-  // TODO: a port cannot have more than SECONDARY_ADDRESS secondary addresses.
+  // TODO: a port cannot have more than MAX_SECONDARY_ADDRESSES secondary addresses.
```

Comment:
This is important.
If would be nice to add a check in the code so that we cannot set more than MAX_SECONDARY_ADDRESSES.